core/src/main/java/org/sonarsource/sonarlint/core/container/global/ExtensionInstaller.java

```diff
@@ -34,24 +34,29 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(ExtensionInstaller.class);
 
-  private final SonarQubeVersion sonarQubeVersion;
+  private final SonarRuntime sqRuntime;
   private final DefaultPluginRepository pluginRepository;
 
-  public ExtensionInstaller(SonarQubeVersion sonarQubeVersion, DefaultPluginRepository pluginRepository) {
-    this.sonarQubeVersion = sonarQubeVersion;
+  public ExtensionInstaller(SonarRuntime sqRuntime, DefaultPluginRepository pluginRepository) {
+    this.sqRuntime = sqRuntime;
     this.pluginRepository = pluginRepository;
   }
 
-  public ExtensionInstaller install(ComponentContainer container, ExtensionMatcher matcher) {
+  public ExtensionInstaller install(ComponentContainer container) {
 
     // plugin extensions
     for (PluginInfo pluginInfo : pluginRepository.getPluginInfos()) {
       Plugin plugin = pluginRepository.getPluginInstance(pluginInfo.getKey());
-      Plugin.Context context = new Plugin.Context(sonarQubeVersion.get());
+      Plugin.Context context = new Plugin.Context(sqRuntime);
       plugin.define(context);
       for (Object extension : context.getExtensions()) {
-        if (!blacklisted(extension)) {
-          doInstall(container, matcher, pluginInfo, extension);
+        Boolean isSlPluginOrNull = pluginInfo.isSonarLintSupported();
+        if (isSlPluginOrNull != null && isSlPluginOrNull.booleanValue()) {
+          if (ExtensionUtils.isSonarLintSide(extension)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)